### ISSUE TITLE
Docker Extension: Can not connect to docker engine when used on SSH remote machine

### ISSUE THREAD

**Issue by qkhanhpro** [microsoft/vscode-remote-release#9386]
I am using VSCode remotely on Ubuntu remote machine using SSH 

Host MyHost
    HostName 18.x.x.x
    User ubuntu
    IdentityFile C:\Users\dqkpr\keys...

On the host there is docker running. 
`docker info `
shows
```
Client: Docker Engine - Community
 Version:    24.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.5
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.7.0
    Path:     /home/ubuntu/.docker/cli-plugins/docker-compose
  scan: Docker Scan (Docker Inc.)
    Version:  v0.23.0
    Path:     /usr/libexec/docker/cli-plugins/docker-scan

Server:
 Containers: 5
  Running: 5
  Paused: 0
  Stopped: 0
 Images: 19
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
 Kernel Version: 5.4.0-1103-aws
 Operating System: Ubuntu 18.04.6 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 1
 Total Memory: 974.2MiB
 Name: ip-172-26-3-161
 ID: H742:C3FI:2PRJ:QJJV:LFUS:ZAF2:H34X:TZ2U:QPLP:G62O:HFMJ:MJLL
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

However the extension seems unable to connect to docker 
![image](https://github.com/microsoft/vscode-remote-release/assets/10443536/d7919d6e-3ec8-4799-bc88-ad4904c23dd2)

If I kill the VSCode process and connect again, docker commands hand indefinetely until I do `unset $DOCKER_HOST`

I use this configuration to connect to multiple different Ubuntu remote host and only 1 machine face this issue

**Comment by joshspicer** [comment]
> If I close the VSCode windows and connect again, docker commands in my terminal hang indefinetely until I do unset $DOCKER_HOST

This sounds like an issue with your docker configuration, and not with the docker extension, if the same thing happens in your terminal until you unset [`DOCKER_HOST`](https://docs.docker.com/reference/cli/docker/#environment-variables).  Do you know where that variable is being set?  Maybe check your ~/.bashrc or similar files on the host?

**Comment by qkhanhpro** [comment]
> > If I close the VSCode windows and connect again, docker commands in my terminal hang indefinetely until I do unset $DOCKER_HOST
> 
> This sounds like an issue with your docker configuration, and not with the docker extension, if the same thing happens in your terminal until you unset [`DOCKER_HOST`](https://docs.docker.com/reference/cli/docker/#environment-variables). Do you know where that variable is being set? Maybe check your ~/.bashrc or similar files on the host?

No. The docker commands such as `docker info` runs fine until I install the docker remote extension. Once the extension is installed, all docker commands fails as well as the docker remote extension hangs.

Once I remove the docker remote extension, I had to `unset DOCKER_HOST` to get my docker commands running normally

**Comment by qkhanhpro** [comment]
Before install docker remote extension
![Image](https://github.com/user-attachments/assets/f11a649b-87be-4f91-adc1-526e47b8ddc1)

After install docker remote extension
![Image](https://github.com/user-attachments/assets/c8aacdd0-1f6d-4550-952a-4fd8618fbe36)

The docker command fail / stuck

Even after removing the docker extension, I need to unset DOCKER_HOST manually to regain access to docker commands

**Comment by joshspicer** [comment]
> until I install the docker remote extension

> I suspect the docker extension tries to set the DOCKER_HOST or something similar, resulting in this issue?

It's sounds like you suspect this is an issue with the [docker extension](https://github.com/Microsoft/vscode-docker). I'll transfer this over to that repo for you to get support.

I could also still see this being some configuration on your remote host interfereing. The SSH extension will not set `DOCKER_HOST` for you automatically, but you may be able to get more hints about what is going on by looking in your SSH log (`> Remote-SSH: Show Log` in the command palette), which will print your remote environment
